modules/workflow-service-remote/src/main/java/org/opencastproject/workflow/remote/WorkflowServiceRemoteImpl.java

```diff
@@ -277,6 +278,73 @@ public WorkflowSet getWorkflowInstancesForAdministrativeRead(WorkflowQuery q) th
     return getWorkflowInstances(q);
   }
 
+  @Override
+  public List<WorkflowInstance> getWorkflowInstancesByMediaPackage(String mediaPackageId)
+          throws WorkflowDatabaseException {
+    List<NameValuePair> queryStringParams = new ArrayList<>();
+
+    queryStringParams.add(new BasicNameValuePair("mediaPackageId", mediaPackageId));
+
+    HttpGet get = new HttpGet("/mediaPackage/" + mediaPackageId + "/instances.xml");
+    HttpResponse response = getResponse(get);
+    try {
+      if (response != null)
+        return XmlWorkflowParser.parseWorkflowSet(response.getEntity().getContent()).getItems();
+    } catch (Exception e) {
+      throw new WorkflowDatabaseException(e);
+    } finally {
+      closeConnection(response);
+    }
+    throw new WorkflowDatabaseException("Workflow instances can not be loaded from a remote workflow service");
+  }
+
+  @Override
+  public Optional<WorkflowInstance> getRunningWorkflowInstanceByMediaPackage(String mediaPackageId, String action)
+          throws WorkflowException, UnauthorizedException, WorkflowDatabaseException {
+    List<NameValuePair> queryStringParams = new ArrayList<>();
+
+    queryStringParams.add(new BasicNameValuePair("mediaPackageId", mediaPackageId));
```

Comment:
same here

Reply:
@lkiesow Maybe I overlooked something, but you didn't meant the string "mediaPackageId" followed by the actual mediaPackageId, did you?

Reply:
I meant exactly that, why?